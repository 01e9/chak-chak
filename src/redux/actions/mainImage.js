export const ACTION_MAIN_IMAGE_SET = 'SET_MAIN_IMAGE'

export function actionMainImageSet(image) {
    return {
        type: ACTION_MAIN_IMAGE_SET,
        image
    }
}
