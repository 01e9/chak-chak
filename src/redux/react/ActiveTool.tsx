import * as React from "react";
import { TOOL_NAME_BRUSH } from "@/constants/tools";
import { connect } from "react-redux";
import { IState } from "@/redux/reducers";

interface IInternalProps {
    activeTool: string;
}

const mapStateToProps = ({ activeTool }: IState) => ({ activeTool });

const render = ({ activeTool }: IInternalProps) => {
    switch (activeTool) {
        case TOOL_NAME_BRUSH: return <div>BRUSH</div>;
        default: return null;
    }
};

export default connect(mapStateToProps)(render);
