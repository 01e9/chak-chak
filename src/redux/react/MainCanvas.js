import * as React from 'react'
import Canvas from "@/react/Canvas";
import { connect } from 'react-redux'
import selectMainImage from "@/redux/selectors/mainImage";
import { ImageStorageConsumer } from "@/react/context/ImageStorage";

const mapStateToProps = (state, ownProps) => ({
    image: selectMainImage(state, ownProps.imageStorage)
})

const MainCanvasConnected = connect(mapStateToProps, null)(Canvas)

const MainCanvas = () => (
    <ImageStorageConsumer>
        {imageStorage => <MainCanvasConnected imageStorage={imageStorage} />}
    </ImageStorageConsumer>
)

export default MainCanvas
