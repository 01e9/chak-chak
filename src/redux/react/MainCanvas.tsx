import * as React from 'react'
import Canvas, { ICanvasProps } from "@/react/Canvas";
import { connect } from 'react-redux'
import selectMainImage from "@/redux/selectors/mainImage";
import { ImageStorageConsumer } from "@/react/context/ImageStorage";
import { IState } from "@/redux/reducers";
import ImageStorage from "@/storage/ImageStorage";

interface IMainCanvasConnectedProps {
    imageStorage: ImageStorage;
}

const mapStateToProps = (state: IState, ownProps: IMainCanvasConnectedProps): (
    Pick<ICanvasProps, "image" | "aspectRatio">
) => ({
    image: selectMainImage(state, ownProps.imageStorage),
    aspectRatio: state.size.aspectRatio,
})

const MainCanvasConnected = connect(mapStateToProps)(Canvas)

const MainCanvas = () => (
    <ImageStorageConsumer>
        {imageStorage => <MainCanvasConnected imageStorage={imageStorage} />}
    </ImageStorageConsumer>
)

export default MainCanvas
