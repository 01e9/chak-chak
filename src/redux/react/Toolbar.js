import * as React from 'react'
import { connect } from 'react-redux'
import Toolbar_ from "@/react/Toolbar";
import { selectToolbarPosition } from "@/redux/selectors/toolbar";
import { actionToolbarPositionSet } from "@/redux/actions/toolbar";
import ToolBrushControl from "@/redux/react/tool/brush/Control";
import { TOOL_NAME_BRUSH } from "@/constants/tools";

const mapStateToProps = (state, ownProps) => {
    const position = selectToolbarPosition(state);

    return {
        left: position.x,
        top: position.y,
        tools: [
            () => <ToolBrushControl key={TOOL_NAME_BRUSH} />
        ]
    }
}

const mapDispatchToProps = (dispatch, ownProps) => ({
    onPositionUpdate: (left, top) => dispatch(actionToolbarPositionSet(left, top))
})

const Toolbar = connect(mapStateToProps, mapDispatchToProps)(Toolbar_)

export default Toolbar
