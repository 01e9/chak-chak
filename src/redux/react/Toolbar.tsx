import * as React from 'react'
import { connect } from 'react-redux'
import Toolbar_, { IToolbarProps } from "@/react/Toolbar";
import { selectToolbarPosition } from "@/redux/selectors/toolbar";
import { actionToolbarPositionSet } from "@/redux/actions/toolbar";
import ToolBrushControl from "@/redux/react/tool/brush/Control";
import { TOOL_NAME_BRUSH } from "@/constants/tools";
import { IState } from "@/redux/reducers";
import { IAction } from "@/redux/utils/actions";
import { Dispatch } from "redux";

const mapStateToProps = (state: IState): Omit<IToolbarProps, "onPositionUpdate"> => {
    const position = selectToolbarPosition(state);

    return {
        left: position.x,
        top: position.y,
        tools: [
            {key: TOOL_NAME_BRUSH, Component: ToolBrushControl}
        ]
    }
}

const mapDispatchToProps = (dispatch: Dispatch<IAction>): Pick<IToolbarProps, "onPositionUpdate"> => ({
    onPositionUpdate: (x, y) => dispatch(actionToolbarPositionSet({x, y}))
})

const Toolbar = connect(mapStateToProps, mapDispatchToProps)(Toolbar_)

export default Toolbar
