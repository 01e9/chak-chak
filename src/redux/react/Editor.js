import Editor_ from "@/react/Editor";
import { connect } from 'react-redux'
import MainCanvas from "@/redux/react/MainCanvas";
import Toolbar from "@/redux/react/Toolbar";

const mapStateToProps = (state, ownProps) => ({
    Canvas: MainCanvas,
    Toolbar: Toolbar,
    ActiveTool: () => '[ActiveTool]' // fixme
})

const Editor = connect(mapStateToProps, null)(Editor_)

export default Editor
