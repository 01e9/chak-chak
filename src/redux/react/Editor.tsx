import * as React from "react";
import Editor, { IEditorProps } from "@/react/Editor";
import { connect } from 'react-redux'
import Canvas from "@/redux/react/MainCanvas";
import Toolbar from "@/redux/react/Toolbar";
import ActiveTool from "@/redux/react/ActiveTool";

const mapStateToProps = (): IEditorProps => ({
    Canvas,
    Toolbar,
    ActiveTool
})

export default connect(mapStateToProps)(Editor)
