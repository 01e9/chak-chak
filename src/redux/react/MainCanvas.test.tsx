import * as React from 'react'
import createReduceRoot from "@/redux/reducers";
import ImageStorage from "@/storage/ImageStorage";
import { actionMainImageSet } from "@/redux/actions/mainImage";
import { createMount } from '@material-ui/core/test-utils';
import MainCanvas from "@/redux/react/MainCanvas";
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { ImageStorageProvider } from "@/react/context/ImageStorage";

const mockStore = configureStore([])

describe('snapshot', () => {
    it('main image', () => {
        const image = new Image(100, 100)
        const imageStorage = new ImageStorage()
        const stateWithMainImage = createReduceRoot(imageStorage)(undefined, actionMainImageSet(image))
        const store = mockStore(stateWithMainImage)

        const wrapper = createMount()(
            <Provider store={store}>
                <ImageStorageProvider value={imageStorage}>
                    <div><MainCanvas/></div>
                </ImageStorageProvider>
            </Provider>
        )
        expect(wrapper.html()).toMatchSnapshot()
    })
})
