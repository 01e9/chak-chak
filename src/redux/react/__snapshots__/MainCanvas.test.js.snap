// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot main image 1`] = `

<div>
  <div class="Canvas_-root-1"
       style="width: 100px; height: 100px;"
  >
  </div>
</div>

`;
